visitor.php

```diff
@@ -796,6 +833,57 @@ private static function getAdditionFromVar(Var_ $tag): ?WordPressTag
         return $tag;
     }
 
+    private static function getTypeNameFromDescription(Description $tagVariableDescription, Type $tagVariableType): ?string
+    {
+        if (!($tagVariableType instanceof \phpDocumentor\Reflection\Types\String_)) {
+            return null;
+        }
+
+        return self::getTypeNameFromDescriptionString($tagVariableDescription->__toString());
+    }
+
+    private static function getTypeNameFromDescriptionString(string $tagDescription = null): ?string
+    {
+        if ($tagDescription === null) {
+            return null;
+        }
+
+        if (str_contains($tagDescription, '@type')) {
+            return null;
+        }
+
+        $fullDescription = str_replace("\n", ' ', $tagDescription);
+
+        /**
+         * This matches phrases that contain a list of two or more single-quoted strings, with the last
+         * item separated by 'or'. The Oxford comma is optional. For example:
+         *
+         * - Either 'am', 'pm', 'AM', or 'PM'
+         * - Accepts 'comment' or 'term'
+         * - Either 'plugin' or 'theme'
+         * - Accepts 'big', or 'little'.
+         * - One of 'default', 'theme', or 'custom'
+         * - Either 'network-active', 'active' or 'inactive'
+         */
+        $matched = preg_match("#(?>returns|either|one of|accepts) ('.+'),? or '([^']+)'#i", $fullDescription, $matches);
```

Comment:
```suggestion
        $matched = preg_match("#(?>returns|either|one of|accepts) ('.+'),? (?>or|and) '([^']+)'#i", $fullDescription, $matches);
```

Reply:
Does `and` catch any additional literals?

Reply:
Yes. For example: [wp_script_is](https://github.com/WordPress/wordpress-develop/blob/6.3/src/wp-includes/functions.wp-scripts.php#L418-L426).
The colon also gives a few more matches.